### DISCUSSION TITLE
Use S3 to store static files

### DISCUSSION THREAD

**Discussion by fkammer** [heyform/heyform#56] [Ideas]
Uploaded files are currently stored locally in `/app/static/upload` and lost when the container is destroyed.

Using a persistent volume works but you're then unable to run multiple heyform instances in parallel for load balancing. Most Kubernetes providers do not support Volumes with `ReadWriteMany` capabilities.

In my opinion the best alternative would be a generic S3 connection.

**Comment by builtcuriously** [comment]
Want to chime in and say there are loads of offers on the market other than AWS such as Clevercloud's Cellar or MinIo and second @fkammer's recommendation for a generic S3 connection.

**Comment by fkammer** [comment]
This feature was added back in April 💯 
https://github.com/heyform/heyform/issues/64
